src/test/java/org/tarantool/MsgPackLiteDeserializeTest.java

```diff
@@ -0,0 +1,475 @@
+            (byte) 0x00, (byte) 0x00, (byte) 0x00, (byte) 0x00,
+        });
+
+        assertEquals(Double.NaN, msgPackLite.unpack(inputStream));
+        assertEquals(Double.NEGATIVE_INFINITY, msgPackLite.unpack(inputStream));
+        assertEquals(Double.POSITIVE_INFINITY, msgPackLite.unpack(inputStream));
+        assertEquals(-213.125d, msgPackLite.unpack(inputStream));
+        assertEquals(100.5d, msgPackLite.unpack(inputStream));
+    }
+
+    /**
+     * Unpacks fixed strings.
+     *
+     * @see MsgPackLiteSerializeTest#testPackFixedStrings() ()
+     */
+    @Test
+    void testUnpackFixedStrings() throws IOException {
+        inputStream = new ByteArrayInputStream(new byte[] {
+            // latin "a"
+            (byte) 0xa1, (byte) 0x61,
+            // latin "hello"
+            (byte) 0xa5, (byte) 0x68, (byte) 0x65, (byte) 0x6c, (byte) 0x6c, (byte) 0x6f,
+            // cyrillic "привет" (12 bytes in utf-8)
+            (byte) 0xac, (byte) 0xd0, (byte) 0xbf, (byte) 0xd1, (byte) 0x80, (byte) 0xd0,
+            (byte) 0xb8, (byte) 0xd0, (byte) 0xb2, (byte) 0xd0, (byte) 0xb5, (byte) 0xd1, (byte) 0x82,
+        });
+
+        assertEquals("a", msgPackLite.unpack(inputStream));
+        assertEquals("hello", msgPackLite.unpack(inputStream));
+        assertEquals("привет", msgPackLite.unpack(inputStream));
+    }
+
+    /**
+     * Unpacks strings.
+     *
+     * @see MsgPackLiteSerializeTest#testPackStrings()
+     */
+    @Test
+    void testUnpackStrings() throws IOException {
+        byte[] oneByteLength = new byte[202];
+        Arrays.fill(oneByteLength, (byte) 'a');
+        oneByteLength[0] = (byte) 0xd9;
+        oneByteLength[1] = (byte) 0xc8;
+        inputStream = new ByteArrayInputStream(oneByteLength);
+        assertEquals(new String(oneByteLength, 2, 200, StandardCharsets.UTF_8), msgPackLite.unpack(inputStream));
+
+        byte[] twoBytesLength = new byte[30003];
+        Arrays.fill(twoBytesLength, (byte) 'b');
+        twoBytesLength[0] = (byte) 0xda;
+        twoBytesLength[1] = (byte) 0x75;
+        twoBytesLength[2] = (byte) 0x30;
+        inputStream = new ByteArrayInputStream(twoBytesLength);
+        assertEquals(new String(twoBytesLength, 3, 30000, StandardCharsets.UTF_8), msgPackLite.unpack(inputStream));
+
+        byte[] fourBytesLength = new byte[80005];
+        Arrays.fill(fourBytesLength, (byte) 'c');
+        fourBytesLength[0] = (byte) 0xdb;
+        fourBytesLength[1] = (byte) 0x00;
+        fourBytesLength[2] = (byte) 0x01;
+        fourBytesLength[3] = (byte) 0x38;
+        fourBytesLength[4] = (byte) 0x80;
+        inputStream = new ByteArrayInputStream(fourBytesLength);
+        assertEquals(new String(fourBytesLength, 5, 80000, StandardCharsets.UTF_8), msgPackLite.unpack(inputStream));
+    }
+
+    /**
+     * Unpacks binaries.
+     *
+     * @see MsgPackLiteSerializeTest#testPackBinArrays()
+     */
+    @Test
+    void testUnpackBinArrays() throws IOException {
```

Comment:
I would add a test case that will deserialize a sequence of bytes that is not a correct UTF-8 to verify that the deserializer does not wrongly assume that data in UTF-8 and does not replace anything with the replacement character. Maybe this is tested implicitly with the large arrays below, but don't sure. Either add a comment that make it obvious or, better, add a small test case that will check it explicitly.